nest/src/game/entities/game.entity.ts

```diff
@@ -103,54 +150,203 @@ export class Game {
     this.status = Status.PENDING;
     this.mode = mode;
   }
-
+  // those are the constants that are used in the game (Henric can adjust them)
   gameConstants = {
-    relativeGameWidth: 1000,
-    relativeMiddle: 500,
-    relativeGameHeight: 1000,
-    player1PaddlePosX: 80,
-    player2PaddlePosX: 920,
-    paddleWidth: 10,
+    relativeGameWidth: 600,
+    relativeMiddleX: 300,
+    relativeGameHeight: 450,
+    relativeMiddleY: 225,
+    player1PaddlePosX: 30,
+    player2PaddlePosX: 570,
+    paddleWidth: 20,
     ballHeight: 30,
-    maxSpeed: 6,
+    maxSpeed: 0,
     speed: 0,
-    speeds: [7, 8, 10, 12, 14, 15, 20],
+    speeds: [5], //[7, 8, 10, 12, 14, 15, 20],
   };
   gameRoomId: string;
   p1id: string | undefined = undefined;
   p2id: string | undefined = undefined;
   status: Status;
   dirx = this.gameConstants.speeds[0];
   diry = 0.0;
-  p1x = 5;
-  p1y = 50;
-  p2x = 95;
-  p2y = 50;
-  bx = 500;
-  by = 55;
+  p1y = 225;
+  p2y = 225;
+  bx = 300;
+  by = 225;
   p1s = 0;
   p2s = 0;
   paddleSize = 100;
   mode: GameMode = GameMode.CLASSIC;
 
+  // Full ball movement logic (moves the ball during the game, and
+  // handles collisions with the game screen and player's paddles.
+  // It also adjusts the speed and direction
+  // of the ball based on the current game mode and the direction of the ball)
+  detectPaddleCollision(
+    ballX: number,
+    ballY: number,
+    ballHeight: number,
+    paddleX: number,
+    paddleY: number,
+    paddleSize: number,
+  ): boolean {
+    const isColliding =
+      ballX === paddleX &&
+      ballY + ballHeight >= paddleY - paddleSize / 2 &&
+      ballY <= paddleY + paddleSize / 2;
+    return isColliding;
+  }
+
+  // Classic Game reset
+  resetBallForClassicMode(): void {
+    const MIN_RANDOM_Y_SPEED = -5;
+    const MAX_RANDOM_Y_SPEED = 5;
+    this.dirx = this.gameConstants.speeds[(this.gameConstants.speed = 0)];
+    this.bx = this.gameConstants.relativeMiddleX;
+    this.by = this.gameConstants.relativeMiddleY;
+    this.diry = generateRandomNumber(MIN_RANDOM_Y_SPEED, MAX_RANDOM_Y_SPEED);
+  }
+
+  //   moveBall(gameService: GameService, server: Server) {
```

Comment:
should we delete that ?

Reply:
When @moabid42 and me were working on that we had some issues with the collision logic. I can delete it for now if you want, but there are going to be more changes in this function in the future anyways.